### PR TITLE
Add DHCP hook to set static hostname after getting from DHCP

### PR DESCRIPTION
**Pull request by taoyl-ms**

### CODE REVIEW

**Comment by lguohan** [comment]
does this work? can you test it?

**Comment by lguohan** [comment]
this make make the `hostname` into /etc/hostname, but who change the hostname in memory?

File: build_debian.sh [hunk 1]
```diff
@@ -239,7 +239,8 @@ allow-hotplug eth0
 iface eth0 inet dhcp
 EOF
 
-sudo cp files/dhcp/rfc3442-classless-routes $FILESYSTEM_ROOT/etc/dhcp/dhclient-exit-hooks.d
+sudo cp files/dhcp/hostname $FILESYSTEM_ROOT/etc/dhcp/dhclient-exit-hooks.d/
+sudo cp files/dhcp/rfc3442-classless-routes $FILESYSTEM_ROOT/etc/dhcp/dhclient-exit-hooks.d/
```
Comment (lguohan) [inline build_debian.sh]: why this line is also changed?

**Comment by taoyl-ms** [comment]
The current hostname needs to be _localhost_ for this to work. When current hostname is _localhost_, dhclient will set the hostname in kernel, and this hook will set the /etc/hostname file. Current default hostname in base image seems to be _sonic_ instead of _localhost_ so it is not working.

**Comment by taoyl-ms** [comment]
On the other hand, #235 will explicitly set both kernel hostname and /etc/hostname upon every dhcpack. So the decision to apply which approach depends on our goal - if we want the hostname be set only once right after deployment, we should adopt this approach, and change hostname in baseimage into _localhost_ instead of current _sonic_. If we want the hostname be able to be updated  even after deployment, we should adopt #235.

File: build_debian.sh (hunk 1 above)
Comment (taoyl-ms) [inline build_debian.sh]: just to add '/' at the end of target to specify it is a folder.

**Comment by lguohan** [comment]
The idea is that when we done with dhcp, we'll disable dhcp and set static IP based on the minigraph, no longer use dhcp any more. so I think #235 fit our need.

**Comment by lguohan** [comment]
let's close this one and comment on #235.

**Pull request closed without merging**